### ISSUE TITLE
[Bug] `An error ocurred during the installation of GiteaPC : Command failed: wsl`

### ISSUE THREAD

**Issue by RedCMD** [yannis300307/CasioDevTools#10]
```An error ocurred during the installation of GiteaPC : Command failed: wsl --shell-type login echo cmd | sudo -S apt install curl git python3 build-essential cmake pkg-config -y
'sudo' is not recognized as an internal or external command,
operable program or batch file.
```

Powershell/CMD runs the 2 commands seperatly rather than passing the 2nd one into wsl
if I add `wsl` just before `sudo` it works perfectly

https://github.com/yannis300307/CasioDevTools/blob/9c44f4d8d3e04334fd972f9a482ea8a7594c8069/src/commands_util.ts#L9

`GiteaPC is now ready to use on your system !`

fxsdk hangs rather than giving an error

**Comment by yannis300307** [comment]
Yes this is an issue with how cmd handles the commands... It works correctly on some computers but not on other... Adding `wsl` before sudo will break the compatibility with Linux systems. I have to find another way to run commands in wsl... 😭
I will try to find a solution the next week-end.

**Comment by yannis300307** [comment]
I am trying to fix this issue. What OS do you use ?

**Comment by RedCMD** [comment]
Windows 11

**Comment by yannis300307** [comment]
I don't understand why you're getting this error because the line which raise your error is not supposed to be executed with windows... Are you running VS code with the WSL extension ?

**Comment by RedCMD** [comment]
I might have been?

when I looked at the code
I saw it had a separate windows and Linux code path
~~and they both merged afterwards~~

**Comment by yannis300307** [comment]
What do you means by "merged" ? Can you send me the line where it happens ?

**Comment by RedCMD** [comment]
> What do you means by "merged"?

sorry I had miss remembered. ignore that

why not just add a Windows/Linux `if` statement?
```js
if (Windows)
    var command = 'echo ' + rootPassword + ' | wsl sudo -S ' + command;
elseif (Linux)
    var command = 'echo ' + rootPassword + ' | sudo -S ' + command;
else
    "error"
```

how can I uninstall GiteaPC to retest this?

I have the WSL extension installed, but I don't remember if it was enabled at the time

**Comment by yannis300307** [comment]
> > What do you means by "merged"?
> 
> sorry I had miss remembered. ignore that
> 
> why not just add a Windows/Linux `if` statement?
> 
> ```js
> if (Windows)
>     var command = 'echo ' + rootPassword + ' | wsl sudo -S ' + command;
> elseif (Linux)
>     var command = 'echo ' + rootPassword + ' | sudo -S ' + command;
> else
>     "error"
> ```

yes I should.

> how can I uninstall GiteaPC to retest this?

If giteapc is already installed, I think you can uninstall it by running `giteapc unistall Lephenixnoir/GiteaPC` but you may have to delete the files manually by deleting all the stuff in the ` ~/.local/share/giteapc` folder. But I think that Lephe could help you better than I am doing.

**Comment by RedCMD** [comment]
```
An error ocurred during the installation of GiteaPC : Command failed: wsl --shell-type login echo pass | sudo -S apt install curl git python3 build-essential cmake pkg-config -y
'sudo' is not recognized as an internal or external command,
operable program or batch file.
```

![image](https://github.com/yannis300307/CasioDevTools/assets/33529441/a7617722-7376-4324-b430-0f0df0723661)

same error with the WSL extension installed or not
![image](https://github.com/yannis300307/CasioDevTools/assets/33529441/3c19e8f7-a357-4d7c-b033-cb0a33ac3086)

**Comment by yannis300307** [comment]
This is strange because `wsl --shell-type loging echo pass | sudo -S ...` should not be executed if you are using Windows...

```typescript
    if (rootPassword) {
        if (OS_NAME === "linux") {
            var command = 'echo ' + rootPassword + ' | sudo -S ' + command;
        } else {
            var command = "-u root " + command;
        }
```

Linux is detected as your OS...
When you go to Help>Toggle Developer Tools and then open the extension, what does it say in the logs?

Personnaly, I have this :
![image](https://github.com/yannis300307/CasioDevTools/assets/52321382/7c8d9e97-465f-45f3-a9c0-b1a34f4eb25a)

**Comment by RedCMD** [comment]
I think installing GiteaPC uses `executeCommand()`, not `executeCommandCallbackOnLog()`
will have to double check
https://github.com/yannis300307/CasioDevTools/blob/9c44f4d8d3e04334fd972f9a482ea8a7594c8069/src/commands_util.ts#L7
https://github.com/yannis300307/CasioDevTools/blob/9c44f4d8d3e04334fd972f9a482ea8a7594c8069/src/commands_util.ts#L39

**Comment by yannis300307** [comment]
I think I have fixed this bug in #c6cb870 but I should add logs on GiteaPC installing.

**Comment by RedCMD** [comment]
I downloaded and built the [another_polish](https://github.com/yannis300307/CasioDevTools/tree/another_polish) branch
GiteaPC failed once (I had to run `sudo apt-get update`), but it then installed fine afterwards
but FxSDK does not

is caused by the same issue with the command not being passed into wsl correctly
https://github.com/yannis300307/CasioDevTools/blob/c6cb870e4824b9abcb096acfac4fe7f1b048302a/src/installations.ts#L132

I couldn't find any error logs.
so I just pasted the command directly into powershell and got this
![image](https://github.com/yannis300307/CasioDevTools/assets/33529441/1a44dbd5-5bbc-473f-85ad-a8a163654a32)

if I paste the command directly into wsl, it works fine
![image](https://github.com/yannis300307/CasioDevTools/assets/33529441/62e7c787-39f9-4950-bae2-c87452f02755)
...
![image](https://github.com/yannis300307/CasioDevTools/assets/33529441/a10c750c-70fa-45f2-ac7e-24405c634b59)

**Comment by yannis300307** [comment]
Normally, `exec()` runs the command in cmd and not in powershell but maybe it's different for you ... I will try to split the installation.

**Comment by yannis300307** [comment]
I split the installation in f41017f45f0b4e186b7ee57849428bc435bee789 is it now working?

**Comment by RedCMD** [comment]
fxsdk hangs and giteapc errors
```
An error ocurred during the installation of GiteaPC : Command failed: wsl --shell-type login cd /tmp/CasioDevToolsGiteaPCInstall; tar -xzf /tmp/CasioDevToolsGiteaPCInstall/giteapc-master.tar.gz


gzip: stdin: not in gzip format
tar: Child returned status 1
tar: Error is not recoverable: exiting now
```
![image](https://github.com/yannis300307/CasioDevTools/assets/33529441/0ab44359-e501-4735-976a-752b0c624a0b)
![image](https://github.com/yannis300307/CasioDevTools/assets/33529441/3eb5025c-113f-4d9f-b76c-237c34471c4c)

**Comment by yannis300307** [comment]
Aaahhh ... I don't understand... Does the error occur immediately after starting the installation?
